bitcoin/script/miniscript.h

```diff
@@ -516,6 +532,26 @@ struct Node {
         return SanitizeType(ComputeType(fragment, x, y, z, sub_types, k, data.size(), subs.size(), keys.size()));
     }
 
+    //! Returns true if the node contains at least one duplicate key.
+    bool ContainsDuplicateKey() const {
+        auto upfn = [](const Node& node, Span<std::set<Key>> subs) -> std::optional<std::set<Key>> {
+            size_t keys_count = node.keys.size();
+
+            std::set<Key> key_set{node.keys.begin(), node.keys.end()};
+            if (key_set.size() != keys_count) return {};
+
+            for (auto& sub: subs) {
+                keys_count += sub.size();
+                key_set.merge(sub);
```

Comment:
One optimization that's possible: put a `if (key_set.size() < sub.size()) std::swap(key_set, sub);` here (`std::set::merge` is linear in the size of the first arg but logarithmic in the size of the second).

Reply:
Also, can use `key_set.merge(std::move(sub))` (avoiding the need to copy the key objects).

Reply:
Done, thanks!